woocommerce/woocommerce-android
# Receipts 3 - printing

Parent issue #4039 

This PR adds support for printing receipts using Android Printing Framework.

The flow is following:
- PaymentSucceeds and ReceiptPaymentInfo is returned from CardReaderManager
- The app shows Payment successful screen with a print button
- User Clicks on the print button
- ViewModel creates ReceiptData using ReceiptDataMapper (unfortunately, ReceiptDataMapper needs to live in the app not in CardReaderModule as it uses WCOrderModel)
- ViewModel creates html receipt using ReceiptCreator
- The html receipt is passed to the view layer (CardReaderPaymentDialog)
- The view layer passes the receipt to a hidden webview which automatically starts the android printing service (using PrintHtmlHelper)

Notes: 
1. The VM also supports "SendReceipt" event, but it's not implemented in the view layer yet (I encountered some issues)
2. Tests for this PR were added in a separate PR to keep the scope of the PRs smaller

To Test:
- Make sure WCPay is installed and activated in test mode on your testing site
1. Open a detail of an unpaid order in USD
2. Tap on "Collect payment" button
3. Go through the connect and payment flows
4. Click on "Print receipt" button
5. Notice a system UI with the receipt is shown
- You can perform other tests to verify that the receipts contains - list of products, discounts, fees, shipping, taxes and total

Update release notes:
